src/main/java/com/flightstats/hub/webhook/WebhookCoordinator.java

```diff
@@ -102,26 +101,45 @@ void ensureRunningOnOnlyOneServer(Webhook daoWebhook, boolean webhookChanged) {
             // associated with a tag webhook
             return;
         }
+
         String name = daoWebhook.getName();
         WebhookLeaderState.RunningState state = webhookLeaderState.getState(name);
-        if (state.isLeadershipAcquired()) {
-            log.debug("found existing webhook {}", name);
-            List<String> servers = new ArrayList<>(state.getRunningServers());
-            if (servers.size() >= 2) {
-                log.warn("found multiple servers leading {}! {}", name, servers);
-                Collections.shuffle(servers);
-                for (int i = 1; i < servers.size(); i++) {
-                    webhookClient.stop(name, servers.get(i));
-                }
-            }
-            if (servers.isEmpty()) {
-                webhookClient.runOnServerWithFewestWebhooks(name);
-            } else if (webhookChanged) {
-                webhookClient.runOnOneServer(name, servers);
-            }
-        } else {
+        WebhookActionDirector director = new WebhookActionDirector(state, webhookChanged);
+
+        if (director.webhookRequiresNoChanges()) {
+            log.debug("no changes required for {}", name);
+        } else if (director.webhookShouldStart()) {
             log.debug("found v2 webhook {}", name);
             webhookClient.runOnServerWithFewestWebhooks(name);
+        } else if (director.webhookShouldRestartOnOneServerAndStopAnyOthers()) {
+            log.debug("found existing webhook {} running on {}", name, state.getRunningServers());
+            webhookClient.runOnOnlyOneServer(name, state.getRunningServers());
+        } else {
+            log.warn("the webhook coordinator seems to have a bug; this statement shouldn't be reached");
+        }
+
+    }
+
+    @VisibleForTesting
+    static class WebhookActionDirector {
```

Comment:
This class has package visibility. Do you still need @VisibleForTesting?

Reply:
Oh, no, I believe I can get rid of it. Thanks!